### PR TITLE
[.NET 5] Rename new SDK, Ref, and Runtime packs

### PR DESCRIPTION
**Pull request by pjcollins**
These changes rename and reorganize the sources and projects used to
create our .NET 5 SDK, Targeting, and Runtime packs.

The `src/Xamarin.Android.Sdk/` directory has been mostly moved to
`src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/`, as it contains
our .NET 5 props and targets files and logic that is very closely tied
to `Xamarin.Android.Build.Tasks` content.

All `.proj` files which were used to create a `nupkg` file have been
moved into `build-tools/create-packs/`.

A new msbuild target has been added to the common `create-packs` targets
file to allow all packs to be created with one simple command:

    msbuild /t:CreateAllPacks build-tools/create-packs/Microsoft.Android.Sdk.proj

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## Unofficial MSBuild Project SDK support
+
+This is an unofficial package which introduces a Xamarin.Android [MSBuild Project SDK](https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk).
+In some ways this is a standalone implementation / extension of MSBuild.Sdk.Extras for Xamarin.Android.
+
+Default Android related file globbing behavior is defined in `Microsoft.Android.Sdk.DefaultItems.props`.
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/README.md:6]: Should this entire file go in this `DotNet5.md`?

https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/DotNet5.md

Then maybe we don't even need a `README.md` here?

**Comment by Nirmal4G** [comment]
I get why, the Sdk project is moved into Build.Tasks project. Can we not put that in the root and gradually assimilate the Build.Tasks into the Sdk project, mirroring what the dotnet-sdk team had done?

**Comment by jonathanpeppers** [comment]
> gradually assimilate the Build.Tasks into the Sdk project, mirroring what the dotnet-sdk team had done?

I don't think dotnet/sdk moved gradually from .NET framework at all, they got to start fresh?

We are keeping the MSBuild targets for .NET 5 along with `Xamarin.Android.Build.Tasks`, because there will be a period of time (probably lengthy) we will support the existing Xamarin.Android codebase & .NET 5.

**Comment by Nirmal4G** [comment]
> there will be a period of time (probably lengthy) we will support the existing Xamarin.Android codebase & .NET 5

I guess that makes sense too. I thought, the Sdk package was the future, may be, we can keep it in the root, to enable better visibility. But, hey, if it helps your workflow then 👌!

File: build-tools/create-packs/Directory.Build.targets [hunk 2]
```diff
@@ -12,20 +12,20 @@
...
+    <_AndroidAppPackAssemblies Include="$(XamarinAndroidSourcePath)bin\$(Configuration)\lib\xamarin.android\xbuild-frameworks\Microsoft.Android\netcoreapp3.1\Mono.Android.Export.dll" />
     <FrameworkListFileClass Include="@(_AndroidAppPackAssemblies->'%(Filename)%(Extension)')" Profile="Android" />
   </ItemGroup>
 
  <!-- https://github.com/dotnet/runtime/blob/0647ec314948904319da5eb15e9931f7c85ed1e2/src/installer/pkg/projects/Directory.Build.targets#L281 -->
   <Target Name="_GenerateFrameworkListFile" >
     <!-- Hardcode framework attributes -->
     <ItemGroup>
-      <FrameworkListRootAttributes Include="Name" Value="Xamarin Android - NET 5.0" />
+      <FrameworkListRootAttributes Include="Name" Value="Android - NET 5.0" />
```
Comment (jonpryor) [inline build-tools/create-packs/Directory.Build.targets:25]: Any guide on correct naming?  Should this be "NET 5.0" or ".NET 5.0"?

Comment (pjcollins) [reply build-tools/create-packs/Directory.Build.targets]: No guide that I am aware aside of existing files in my `dotnet` install... I've updated this to be named `.NET 5.0 - Android` instead.

**Pull request merged**